### DISCUSSION TITLE
Add possibility to use DevTools

### DISCUSSION THREAD

**Discussion by viganzeqiri** [wevm/wagmi#1228] [Ideas]
Feature request to add devtools in order to make it easier to manage the smart contract cached data on dev mode.
https://tanstack.com/query/v4/docs/devtools

**Comment by nejcm** [comment]
Second this. It is a very useful tool while testing and developing and very easy to integrate.

**Comment by Yuripetusko** [comment]
I had a bigint serialization error when running tanstack query devtools, but was able to solve it by passing the custom query key function to the query client


```
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      queryKeyHashFn: (queryKey: QueryKey): string => {
        return JSON.stringify(queryKey, (_, value) => {
          if (isPlainObject(value))
            return Object.keys(value)
              .sort()
              .reduce((result, key) => {
                result[key] = value[key];
                return result;
              }, {} as any);
          if (typeof value === "bigint") return value.toString();
          return value;
        });
      },
    },
  },
});
```

**Reply by Yuripetusko** [reply]
@tmm do you think it's worth exporting your `queryKeyHashFn` function that handles bigint serialization, and then adding this snipper above to the docs on wagmi's TanStack Query page in a new "Debugger" section? If so, then I can create a pr

**Reply by tmm** [reply]
Go for it
